### ISSUE TITLE
Flex-grow doesn't work

### ISSUE THREAD

**Issue by DougRzz** [jupyter-widgets/ipywidgets#1978]
Hi,  I am trying to understand the use of flex-grow in the layouts of Jupyter Widgets and I think I have found a bug.  In the documentation example in http://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Styling.html?highlight=Layout%2C%20Button , the middle button should be twice the size of the outer buttons.

![image](https://user-images.githubusercontent.com/3510318/36357664-2833ca5a-14f9-11e8-96f7-04d554249510.png)

```python
from ipywidgets import Layout, Button, Box

items = [
    Button(description='weight=1'),
    Button(description='weight=2', layout=Layout(flex='2 1 auto', width='auto')),
    Button(description='weight=1'),
 ]

box_layout = Layout(display='flex',
                    flex_flow='row',
                    align_items='stretch',
                    border='solid',
                    width='50%')
box = Box(children=items, layout=box_layout)
box
```
Removing the "flex='2 1 auto'"  has no effect on the size of the buttons.

**Comment by jasongrout** [comment]
> the middle button should be twice the size of the outer buttons.

I think the confusion is that's not what the text "items flex proportional to their weight" means. I'd encourage you to read the flexbox tutorial pointed out in that documentation page (https://css-tricks.com/snippets/css/a-guide-to-flexbox/), as it's a bit subtle. Basically, it's going to take the space left over by the widgets and divide that leftover space in proportion to the weights. It's not about stretching the widgets to be twice the size of other widgets.

**Comment by DougRzz** [comment]
@jasongrout Thanks for the explanation.  It got me thinking and I think I have figured out the problem.  The other 2 items needed the flex grow to be specified - that is why the flex='2 1 auto' wasn't actually doing anything (as incorrectly shown in the 2nd example here: http://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Styling.html#Examples).  I will submit a PR later in the week to suggest a documentation change.    

![image](https://user-images.githubusercontent.com/3510318/37342656-21f0316c-26bd-11e8-9b9e-5a5e357e0a59.png)

**Comment by jasongrout** [comment]
> I will submit a PR later in the week to suggest a documentation change.

Great, thanks!

**Comment by jasongrout** [comment]
Fixed by #2010, which is a rebased version #2001.
